packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -393,13 +394,19 @@ module.exports = function(
 
     if (instance.state !== oldState) {
       if (__DEV__) {
+        const currentComponent =
+          getComponentName(workInProgress) || 'Component';
+        if (stateDeprecationWarning[currentComponent]) {
+          return;
```

Comment:
We shouldn't return here since that won't execute the code outside of DEV. Let's just wrap the warning into a condition.